### PR TITLE
Simplify block assembly.

### PR DESCRIPTION
**Pull request by derbear**
This change simplifies and slightly speeds up block assembly.

This change moves block assembly to the transaction pool in preparation for further optimizations, which involve coordinating between the block assembly goroutine and the transaction pool to reduce wasted work.

### CODE REVIEW

File: node/node.go [hunk 1]
```diff
@@ -231,7 +230,7 @@ func MakeFull(log logging.Logger, rootDir string, cfg config.Local, phonebookAdd
 		Local:          node.config,
 		Network:        gossip.WrapNetwork(node.net, log),
 		Ledger:         agreementLedger,
-		BlockFactory:   blockFactory,
+		BlockFactory:   blockFactory{TransactionPool: node.transactionPool},
```
Comment (tsachiherman) [inline node/node.go:233]: If you're doing this, why don't you eliminate the blockFactoryImpl completely and move it's functionality to the node ?

Comment (derbear) [reply node/node.go]: Sounds good--removed.

File: data/pools/transactionPool.go [hunk 2]
```diff
@@ -555,3 +559,133 @@ func (pool *TransactionPool) recomputeBlockEvaluator(committedTxIds map[transact
...
+	stats.Nanoseconds = dt.Nanoseconds()
+
+	if pool.logAssembleStats {
+		payset := lvb.Block().Payset
+		if len(payset) != 0 {
+			totalFees := uint64(0)
+
+			for i, txib := range payset {
+				fee := txib.Txn.Fee.Raw
+				encodedLen := txib.GetEncodedLength()
```
Comment (algobolson) [inline data/pools/transactionPool.go:655]: I kinda bet that this is the slowest part of turning on logAssembleStats; maybe a TODO comment or a backlog issue for making a stats-lite mode that records the cheap stats?

**Comment by derbear** [comment]
@egieseke ran one set of S1 tests on this branch tip, which showed a 2% decrease in throughput relative to master. Any opinions on whether this merits holding off on a merge?

**Pull request merged**